lib/src/main/scala/spinal/lib/bus/regif/BusIfAdapter/BRAMBusInterface.scala

```diff
@@ -0,0 +1,43 @@
+package spinal.lib.bus.regif
+
+import spinal.core._
+import spinal.lib.bus.misc.SizeMapping
+import spinal.lib.bus.bram.BRAM
+import spinal.lib.Delay
+
+case class BRAMBusInterface(bus: BRAM, sizeMap: SizeMapping, regPre: String = "", readDelay: Int = 1)(implicit moduleName: ClassName) extends BusIf {
+  override val withStrb: Boolean = true
+  override val wstrb: Bits = withStrb generate(Bits(strbWidth bit))
+  override val wmask: Bits = withStrb generate(Bits(busDataWidth bit))
+  override val wmaskn: Bits = withStrb generate(Bits(busDataWidth bit))
+  withStrb generate(wstrb := bus.we)
+  initStrbMasks()
+
+  override val askWrite: Bool = bus.we.orR.allowPruning()
+
+  override val askRead: Bool = (!askWrite).allowPruning()
+
+  override val doWrite: Bool = (askWrite && bus.en).allowPruning()
+
+  override val doRead: Bool = (askRead && bus.en).allowPruning()
+
+  override val readData: Bits = Bits(busDataWidth bits)
+
+  bus.rddata := Delay(readData, readDelay) init B(0)
```

Comment:
I may have to check again, but looking at the BRAM bundle I don't see how arbitrary delay can work with the bus - since there is no response signal in the bus to indicate when a response is available?

Reply:
I wouldn't call it an arbitrary delay, it is more so that with BRAM you always get a response after a constant number of cycles, depending on how it is configured. So the register interface must match the constant read delay of the BRAM.

Perhaps I should make this a configuration option in the bus itself?

Reply:
The other components that deal with the BRAM bus also can't deal with arbitrary read delay from the BRAM (Decoder, BRAM BusSlaveFactory and Axi4SharedToBram depend on a 1 cycle delay). That's why I'm hesitant w.r.t. adding this.

Reply:
I have moved the setting to BRAMConfig, as I think it is the cleanest solution. Then, to make sure the rest of the components are compatible:
- An assert when connecting two BRAM buses to make sure they have the same read latency.
- Decoder, RegIf and BRAMDriver support any read latency.
- BusSlave has an assert preventing read latency other than 1.
- Axis4SharedToBram creates its own BRAM config, so it will always have an implicit read latency of 1.

Reply:
Looks good for most of this, but I don't see how the BRAMDecoder would support any latency? If you are accessing different RAM blocks every cycle, then the `selIndex` needs to be delayed for the number of read cycles, otherwise you'd mux the data from the wrong RAM to the `rddata`?

Reply:
Yes you are right. I remember making that change, but perhaps I only thought about it.

I will make the edit after jijingg responds and we decide what to do with his commit.